modules/tinymce/src/core/main/ts/api/util/ImageUploader.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Fun } from '@ephox/katamari';
+import * as Uploader from '../../file/Uploader';
+import { UploadStatus } from '../../file/UploadStatus';
+import Editor from '../Editor';
+import * as Settings from '../Settings';
+import { BlobInfo } from '../file/BlobCache';
+
+export type UploadResult = Uploader.UploadResult;
+
+interface ImageUploader {
+  upload (blobInfos: BlobInfo[], showNotification?: boolean): Promise<UploadResult[]>;
+}
+
+export const openNotification = (editor: Editor) => () => editor.notificationManager.open({
+  text: editor.translate('Image uploading...'),
+  type: 'info',
+  timeout: -1,
+  progressBar: true
+});
+
+export const createUploader = (editor: Editor, uploadStatus: UploadStatus) =>
+  Uploader.Uploader(uploadStatus, {
+    url: Settings.getImageUploadUrl(editor),
+    basePath: Settings.getImageUploadBasePath(editor),
+    credentials: Settings.getImagesUploadCredentials(editor),
+    handler: Settings.getImagesUploadHandler(editor)
+  });
+
+/**
+ * This class handles uploading images to a back-end server.
```

Comment:
Sorry, @wendelcosta I forgot we need to include what version this was added in, so we'll need something like this:
```suggestion
 * This class handles uploading images to a back-end server.
 * <br>
 * <em>Added in TinyMCE 5.7</em>
```

Reply:
Done